src/main/java/io/github/thebusybiscuit/slimefun4/api/events/ExplosiveToolBreakBlockEvent.java

```diff
@@ -0,0 +1,83 @@
+package io.github.thebusybiscuit.slimefun4.api.events;
+
+import io.github.thebusybiscuit.slimefun4.implementation.items.tools.ExplosiveTool;
+import org.bukkit.block.Block;
+import org.bukkit.entity.Player;
+import org.bukkit.event.Cancellable;
+import org.bukkit.event.Event;
+import org.bukkit.event.HandlerList;
+import org.bukkit.event.player.PlayerEvent;
+import org.bukkit.inventory.ItemStack;
+
+import javax.annotation.Nonnull;
+import javax.annotation.ParametersAreNonnullByDefault;
+import java.util.List;
+
+/**
+ * This {@link Event} is called when a {@link Block} is destroyed by an {@link ExplosiveTool}.
+ *
+ * @author GallowsDove
+ *
+ */
+public class ExplosiveToolBreakBlockEvent extends PlayerEvent implements Cancellable {
+
+    private static final HandlerList handlers = new HandlerList();
+
+    private final ItemStack tool;
+    private final ExplosiveTool explosiveTool;
+    private final List<Block> blocks;
+    private boolean cancelled;
+
+    @ParametersAreNonnullByDefault
+    public ExplosiveToolBreakBlockEvent(Player player, List<Block> blocks, ItemStack item, ExplosiveTool explosiveTool) {
+        super(player);
```

Comment:
The parameters are marked as nonnull, so you should add `Validate.notNull(x, msg)` validations here.